### PR TITLE
[MXNET-1400]adding large tensor support tests for depth_to_space and space_to_depth

### PR DESCRIPTION
**Pull request by access2rohit**
## Description ##
This PR adds large tensor test cases for 2 more ops mentioned in the title. 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by access2rohit** [comment]
@mxnet-label-bot add [pr-awaiting-review]

**Comment by access2rohit** [comment]
@apeforest : please review

File: tests/nightly/test_large_array.py [hunk 1]
```diff
@@ -185,7 +186,19 @@ def test_pick():
     b = mx.nd.ones(shape=(256*35,))
     res = mx.nd.pick(a,b)
     assert res.shape == b.shape
-
```
Comment (apeforest) [inline tests/nightly/test_large_array.py]: keep this line.

File: tests/nightly/test_large_array.py [hunk 2]
```diff
@@ -185,7 +186,19 @@ def test_pick():
...
+    
+def test_depthtospace():
+    shape_inp = (LARGE_X, 8, 4, 2)
+    data = rand_ndarray(shape_inp, 'default')
+    data_np = data.asnumpy()
+    output = mx.nd.depth_to_space(data, 2)
+
```
Comment (apeforest) [inline tests/nightly/test_large_array.py]: add extra line to follow pep8 style guide

File: tests/nightly/test_large_array.py [hunk 3]
```diff
@@ -185,7 +186,19 @@ def test_pick():
...
+def test_spacetodepth():
+    shape_inp = (LARGE_X, 2, 8, 4)
+    data = rand_ndarray(shape_inp, 'default')
+    data_np = data.asnumpy()
+    output = mx.nd.space_to_depth(data, 2)
+    mx.nd.waitall()
```
Comment (apeforest) [inline tests/nightly/test_large_array.py]: add extra line to follow pep8 style guide

File: tests/nightly/test_large_array.py [hunk 4]
```diff
@@ -185,7 +186,19 @@ def test_pick():
...
     assert res.shape == b.shape
-
+    
+def test_depthtospace():
+    shape_inp = (LARGE_X, 8, 4, 2)
+    data = rand_ndarray(shape_inp, 'default')
+    data_np = data.asnumpy()
+    output = mx.nd.depth_to_space(data, 2)
+
+def test_spacetodepth():
```
Comment (apeforest) [inline tests/nightly/test_large_array.py]: what are you testing here?

File: tests/nightly/test_large_array.py [hunk 5]
```diff
@@ -185,7 +186,19 @@ def test_pick():
     b = mx.nd.ones(shape=(256*35,))
     res = mx.nd.pick(a,b)
     assert res.shape == b.shape
-
+    
+def test_depthtospace():
```
Comment (apeforest) [inline tests/nightly/test_large_array.py]: what are you testing here?

File: tests/nightly/test_large_array.py [hunk 6]
```diff
@@ -185,7 +186,19 @@ def test_pick():
...
+    data = rand_ndarray(shape_inp, 'default')
+    data_np = data.asnumpy()
+    output = mx.nd.depth_to_space(data, 2)
+
+def test_spacetodepth():
+    shape_inp = (LARGE_X, 2, 8, 4)
+    data = rand_ndarray(shape_inp, 'default')
+    data_np = data.asnumpy()
+    output = mx.nd.space_to_depth(data, 2)
+    mx.nd.waitall()
```
Comment (apeforest) [inline tests/nightly/test_large_array.py]: no need waitall here. You can use output.asnumpy()

File: tests/nightly/test_large_array.py (hunk 1 above)
Comment (access2rohit) [reply tests/nightly/test_large_array.py]: done

File: tests/nightly/test_large_array.py (hunk 2 above)
Comment (access2rohit) [reply tests/nightly/test_large_array.py]: done

File: tests/nightly/test_large_array.py (hunk 4 above)
Comment (access2rohit) [reply tests/nightly/test_large_array.py]: Have added test cases to compare with numpy implementation

File: tests/nightly/test_large_array.py [hunk 7]
```diff
@@ -185,7 +186,36 @@ def test_pick():
     b = mx.nd.ones(shape=(256*35,))
     res = mx.nd.pick(a,b)
     assert res.shape == b.shape
-
+    
+def test_depthtospace():
+    def f(x, blocksize):
```
Comment (apeforest) [inline tests/nightly/test_large_array.py]: What does this method do? Use a meaningful name.

File: tests/nightly/test_large_array.py [hunk 8]
```diff
@@ -185,7 +186,36 @@ def test_pick():
...
+
+    shape_inp = (LARGE_X, 8, 4, 2)
+    data = rand_ndarray(shape_inp, 'default')
+    data_np = data.asnumpy()
+    expected = f(data_np, 2)
+    output = mx.nd.depth_to_space(data, 2)
+    assert_almost_equal(output.asnumpy(), expected, atol=1e-3, rtol=1e-3)
+
+def test_spacetodepth():
+    def f(x, blocksize):
```
Comment (apeforest) [inline tests/nightly/test_large_array.py]: What does this method do? Use a meaningful name.

File: tests/nightly/test_large_array.py (hunk 5 above)
Comment (access2rohit) [reply tests/nightly/test_large_array.py]: Have added test cases to compare with numpy implementation

File: tests/nightly/test_large_array.py (hunk 3 above)
Comment (access2rohit) [reply tests/nightly/test_large_array.py]: done

File: tests/nightly/test_large_array.py (hunk 8 above)
Comment (access2rohit) [reply tests/nightly/test_large_array.py]: I have updated the names of the primitives

File: tests/nightly/test_large_array.py (hunk 7 above)
Comment (access2rohit) [reply tests/nightly/test_large_array.py]: I have updated the names of the function

File: tests/nightly/test_large_array.py [hunk 9]
```diff
@@ -17,6 +17,7 @@
 
 import mxnet as mx
 import numpy as np
+from mxnet.test_utils import rand_ndarray, assert_almost_equal
```
Comment (apeforest) [inline tests/nightly/test_large_array.py]: nit: can we change to `from mxnet import test_utils` to increase readability?

File: tests/nightly/test_large_array.py (hunk 5 above)
Comment (apeforest) [reply tests/nightly/test_large_array.py]: nit: rename to `test_depth_to_space`

File: tests/nightly/test_large_array.py [hunk 10]
```diff
@@ -185,7 +186,36 @@ def test_pick():
     b = mx.nd.ones(shape=(256*35,))
     res = mx.nd.pick(a,b)
     assert res.shape == b.shape
-
+    
+def test_depthtospace():
+    def depthtospace_using_numpy(x, blocksize):
```
Comment (apeforest) [inline tests/nightly/test_large_array.py]: nit: rename to `numpy_depth_to_space`

File: tests/nightly/test_large_array.py [hunk 11]
```diff
@@ -185,7 +186,36 @@ def test_pick():
...
+
+    shape_inp = (LARGE_X, 8, 4, 2)
+    data = rand_ndarray(shape_inp, 'default')
+    data_np = data.asnumpy()
+    expected = depthtospace_using_numpy(data_np, 2)
+    output = mx.nd.depth_to_space(data, 2)
+    assert_almost_equal(output.asnumpy(), expected, atol=1e-3, rtol=1e-3)
+
+def test_spacetodepth():
+    def spacetodepth_using_numpy(x, blocksize):
```
Comment (apeforest) [inline tests/nightly/test_large_array.py]: nit: rename to `numpy_space_to_depth`

**Comment by access2rohit** [comment]
@mxnet-label-bot add [pr-awaiting-merge]

**Pull request merged**